### PR TITLE
v0.1 components

### PR DESCRIPTION
**Pull request by mrhouzlane**
**V0.1 Components** : Identity, Effect, Power, Position, OwnedBy, isRevealed,  _GameConfigComponent_.
Last component is a TO_DO task after figuring out system logic.

### CODE REVIEW

**Comment by Oba-One** [comment]
> **V0.1 Components** : Identity, Effect, Power, Position, OwnedBy, isRevealed, _GameConfigComponent_. Last component is a TO_DO task after figuring out system logic.

Awesome, can you add the new components to the architecture doc

File: packages/contracts/src/libraries/DSTypes.sol [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.0;
+
+import "std-contracts/components/CoordComponent.sol";
```
Comment (junaama) [inline packages/contracts/src/libraries/DSTypes.sol]: `CoordComponent` is unused and it's not needed for any of our types

File: packages/contracts/src/libraries/DSTypes.sol [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+struct GameConfig {
+    // Block timestamp when the game started
+    uint256 startTime;
+    // Number of seconds
+    uint32 commitPhaseLength;
+    uint32 revealPhaseLength;
+    uint32 actionPhaseLength;
+    uint32 worldRadius;
```
Comment (junaama) [inline packages/contracts/src/libraries/DSTypes.sol]: Why is `worldRadius` needed?

File: packages/contracts/src/components/EffectComponent.sol [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "std-contracts/components/Uint32Component.sol";
+
+uint256 constant ID = uint256(keccak256("component.Counter"));
```
Comment (junaama) [inline packages/contracts/src/components/EffectComponent.sol]:
```suggestion
uint256 constant ID = uint256(keccak256("component.Effect"));
```

File: packages/contracts/src/components/GameConfigComponent.sol [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        return getEntitiesWithValue(encodedValue(config));
+    }
+
+    function encodedValue(GameConfig calldata config) private pure returns (bytes memory) {
+        return abi.encode( // TODO : change names of tuple params
+            config.startTime,
+            config.commitPhaseLength,
+            config.revealPhaseLength,
+            config.actionPhaseLength,
+            config.worldRadius
```
Comment (junaama) [inline packages/contracts/src/components/GameConfigComponent.sol]:
```suggestion
```

File: packages/contracts/src/components/IdentityComponent.sol [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.0;
+
+import "std-contracts/components/StringComponent.sol";
+
+uint256 constant ID = uint256(keccak256("ds.component.Name"));
```
Comment (junaama) [inline packages/contracts/src/components/IdentityComponent.sol]: `ds.` is what Dark Seas used to differentiate their world component names - let's just be sure to remove it before pushing

File: packages/contracts/src/components/OwnedByComponent.sol [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+pragma solidity >=0.8.0;
+
+import "std-contracts/components/BoolComponent.sol";
+
+uint256 constant ID = uint256(keccak256("ds.component.Loaded"));
+
+///@title OwnedBy for item Character
+///@author Mehdi R.
+///@notice You can deploy this contract as ID for a system
+contract OwnedBy is BoolComponent {
```
Comment (junaama) [inline packages/contracts/src/components/OwnedByComponent.sol]: OwnedBy shouldnt be a boolean, what we want here is the uint256 ID corresponding to a component, so it should be Uint256

File: packages/contracts/src/components/PowerComponent.sol [hunk 7]
```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "solecs/Component.sol";
```
Comment (junaama) [inline packages/contracts/src/components/PowerComponent.sol]:
```suggestion
import { Uint32Component } from "std-contracts/components/Uint32Component.sol";
```

File: packages/contracts/src/libraries/LibCardMove.sol [hunk 8]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+//External
+import {getAddressById} from "solecs/utils.sol";
+import {IUint256Component} from "solecs/interfaces/IUint256Component.sol";
+
+//Components
+import {PositionComponent, ID as PositionComponentID} from "../components/Position.sol";
+import {Uint32Component} from "std-contracts/components/Uint32Component.sol";
```
Comment (junaama) [inline packages/contracts/src/libraries/LibCardMove.sol:10]: nit: This can move up under `// External`

**Review by junaama — APPROVED** [review]

**Pull request merged**